client/app/license/redirection/upgrade/license-redirection-upgrade.controller.js

```diff
@@ -1,12 +1,21 @@
-angular.module("Module.license").controller("ConfirmUpgradeRedirectionCtrl", ($location, $scope, $timeout) => {
-    "use strict";
+angular.module("Module.license").controller("LicenseRedirectionUpgradeCtrl", class LicenseRedirectionUpgradeCtrl {
+    constructor ($scope, $state, $timeout) {
+        this.$scope = $scope;
+        this.$state = $state;
+        this.$timeout = $timeout;
+    }
 
-    const licenseId = $scope.currentActionData.licenseId;
+    $onInit () {
+        this.licenseId = this.$scope.currentActionData.licenseId;
+        this.$scope.redirectToUpgrade = this.redirectToUpgrade.bind(this);
+    }
 
-    $scope.redirectToUpgrade = function () {
-        $scope.resetAction();
-        $timeout(() => {
-            $location.path(`/configuration/license/upgrade/${licenseId}`);
+    redirectToUpgrade () {
+        this.$scope.resetAction();
+        this.$timeout(() => {
```

Comment:
why 300 ms timeout ?

Reply:
why `$timeout` ?

Reply:
Don't look at me as if I was guilty :p

Reply:
You could have just removed it when you reached it 😃

But we know you didn't write it down to begin with !

Reply:
The timeout prevents the modal backdrop to stay here on the next page. :(

Reply:
Ok! So let's replace with a stateless modal in a separate pull request.